### PR TITLE
docs: create concepts docs page and diagram

### LINKED ISSUE #3802: Concepts page

### PR DESCRIPTION
**Pull request by missBerg**
**What this PR does / why we need it**:
Includes a Concept page for the relevant resources for the docs site, with an accompanying diagram to show the relationship between the resources.

The page is supposed to help people understand the relationship between the resources.

**Which issue(s) this PR fixes**:
Fixes #3802

**Note:** I changed this because I messed up with signing the commits on the other one.

### CODE REVIEW

File: site/content/en/latest/concepts/concepts_overview.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Kubernetes Gateway API Resources
+- **GatewayClass:** Defines a class of Gateways with common configuration.
+- **Gateway:** Specifies how traffic can enter the cluster.
+- **Routes:** **HTTPRoute, GRPCRoute, TLSRoute, TCPRoute, UDPRoute:** Define routing rules for different types of traffic.
+## Envoy Gateway (EG) API Resources
+- **EnvoyProxy:** Represents the deployment and configuration of the Envoy proxy within a Kubernetes cluster, managing its lifecycle and settings.
+- **EnvoyPatchPolicy, ClientTrafficPolicy, SecurityPolicy, BackendTrafficPolicy, EnvoyExtensionPolicy, BackendTLSPolicy:** Additional policies and configurations specific to Envoy Gateway.
+
+| API         | Resource                                       | Required | Purpose            | References           | Description                                                                                                                                                                                                 |
```
Comment (arkodg) [inline site/content/en/latest/concepts/concepts_overview.md:15]: can we swap the position of the Resource column and make it the first one ?

File: site/content/en/latest/concepts/concepts_overview.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+| ----------- | ---------------------------------------------- | -------- | ------------------ | -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Gateway API | GatewayClass                                   | Yes      | Gateway Config     | Core                 | Defines a class of Gateways with common configuration.                                                                                                                                                      |
```
Comment (arkodg) [inline site/content/en/latest/concepts/concepts_overview.md:17]: non blocking comment: would be great to add hyperlinks to the resources in a follow up PR, Gateway API links blocked on https://github.com/envoyproxy/gateway/issues/2749

**Comment by arkodg** [comment]
hey @missBerg can you also copy paste the same content into the `site/content/en/v1.0.2/concepts/` dir ?

File: site/content/en/latest/concepts/concepts_overview.md (hunk 2 above)
Comment (missBerg) [reply site/content/en/latest/concepts/concepts_overview.md:17]: Yeah I agree that would be good

File: site/content/en/latest/concepts/concepts_overview.md (hunk 1 above)
Comment (missBerg) [reply site/content/en/latest/concepts/concepts_overview.md:15]: Asking for me to move mountains haha. It should be doable

Comment (missBerg) [reply site/content/en/latest/concepts/concepts_overview.md]: Done!

**Comment by zhaohuabing** [comment]
Should we also mention the EG Backend API(representing non-k8s backend services) in the docs and diagram?

**Comment by arkodg** [comment]
> Great job! @missBerg Just one thing, should we also include the EG Backend API(representing non-k8s backend services) in the docs and diagram?

Thanks for catching this @zhaohuabing, @missBerg hopefully this is the last update :see_no_evil

**Comment by missBerg** [comment]
hopefully it is good now

**Review by arkodg — APPROVED** [review]
LGTM thanks for updating the diagram !

**Comment by zirain** [comment]
```console
[404] site/public/v1.0.2/concepts/tasks/traffic/backend
```

**Comment by Xunzhuo** [comment]
We should move 1.0.2 to 1.0, since we do not keep minor version docs path.

**Review by Xunzhuo — APPROVED** [review]
Helped fix the lint and some other issues, BTW, we should only add this to /latest for now, and cut off when releasing v1.1.

**Comment by missBerg** [comment]
Thanks @Xunzhuo ❤️

**Review by zirain — APPROVED** [review]

**Pull request merged**